hazelcast/util.py

```diff
@@ -272,18 +276,65 @@ def next(self):
         """
         raise NotImplementedError("next")
 
+    def next_data_member(self):
+        """Returns the next data member to route to.
+
+        Returns:
+            hazelcast.core.MemberInfo: The next data member or
+            ``None`` if no data member is available.
+        """
+        return None
+
+    def can_get_next_data_member(self):
+        """Returns whether this instance supports getting data members
+         through a call to :func:`next_data_member`.
+
+        Returns:
+            bool: ``True`` if this instance supports getting data members.
+        """
+        return False
+
+
+class _Members(object):
+    __slots__ = ("members", "data_members")
+
+    def __init__(self, members, data_members):
+        self.members = members
+        self.data_members = data_members
+
 
 class _AbstractLoadBalancer(LoadBalancer):
     def __init__(self):
         self._cluster_service = None
-        self._members = []
+        self._members = _Members([], [])
 
     def init(self, cluster_service):
         self._cluster_service = cluster_service
         cluster_service.add_listener(self._listener, self._listener, True)
 
+    def next(self):
+        members = self._members.members
+        return self._next(members)
+
+    def next_data_member(self):
+        members = self._members.data_members
+        return self._next(members)
+
+    def can_get_next_data_member(self):
+        return True
+
     def _listener(self, _):
-        self._members = self._cluster_service.get_members()
+        members = self._cluster_service.get_members()
+        data_members = []
+
+        for member in members:
+            if not member.lite_member:
+                data_members.append(member)
```

Comment:
you could simplify here

```suggestion
        data_members = list(filter(lambda member: not member.lite_member, members))
```